helix-term/src/commands/typed.rs

```diff
@@ -1897,48 +1797,61 @@ fn get_option(
 
 /// Change config at runtime. Access nested values by dot syntax, for
 /// example to disable smart case search, use `:set search.smart-case false`.
-fn set_option(
-    cx: &mut compositor::Context,
-    args: &[Cow<str>],
-    event: PromptEvent,
-) -> anyhow::Result<()> {
+fn set_option(cx: &mut compositor::Context, args: Args, event: PromptEvent) -> anyhow::Result<()> {
     if event != PromptEvent::Validate {
         return Ok(());
     }
 
-    if args.len() != 2 {
-        anyhow::bail!("Bad arguments. Usage: `:set key field`");
-    }
-    let (key, arg) = (&args[0].to_lowercase(), &args[1]);
+    let mut parser = args.raw_parser();
```

Comment:
To remove the remaining `raw_parser`s you can set `:set-option` and `:toggle-option` as `ParseMode::Literal` and instead here say:

```rust
let mut parser = shellwords::ArgParser::from(args[0]);
```